include/drivers/spi.h

```diff
@@ -363,6 +417,28 @@ __subsystem struct spi_driver_api {
 	spi_api_release release;
 };
 
+/**
+ * @brief Validate that SPI bus is ready.
+ *
+ * @param spec SPI specification from devicetree
+ *
+ * @retval true if the SPI bus is ready for use.
+ * @retval false if the SPI bus is not ready for use.
+ */
+static inline bool spi_is_ready(const struct spi_dt_spec *spec)
+{
+	/* Validate bus is ready */
+	if (!device_is_ready(spec->bus)) {
+		return -EINVAL;
```

Comment:
I think we usually use ENODEV for devices that are not ready.

Reply:
From a quick browse it looks like about a 50/50 split between ENODEV and EINVAL in the drivers.
A convention should probably be established in another PR.